### PR TITLE
dep types: Use TsFlower for expo-application and expo-screen-orientation

### PR DESCRIPTION
**Pull request by chrisbobbe**
I'd like to do this on the way to #5441, upgrading to Expo SDK 44.

### CODE REVIEW

File: types/patches/0027-tsflower-expo-application-Translate-a-TS-enum.patch [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+-    SIMULATOR = 1,
+-    ENTERPRISE = 2,
+-    DEVELOPMENT = 3,
+-    AD_HOC = 4,
+-    APP_STORE = 5
+-} */
++declare export var ApplicationReleaseType: {|
++  +UNKNOWN: 0,
++  +SIMULATOR: 1,
++  +ENTERPRISE: 2,
```
Comment (gnprice) [inline types/patches/0027-tsflower-expo-application-Translate-a-TS-enum.patch:64]: Would be good to mention that this doesn't describe all the quirky behavior of these TS enums (namely that it maps values to names as well as names to values.)

**Review by gnprice — COMMENTED** [review]
Thanks @chrisbobbe! This looks good -- small comments below.

**Comment by gnprice** [comment]
Oh also: the references to `ft-flow` in eslint-ignore lines look like they depend on part of #5444 .

**Comment by chrisbobbe** [comment]
Thanks for the review! Revision pushed. I removed the `ft-flow` eslint-ignore lines; it seems ESLint isn't being run on these files anyway.

**Comment by gnprice** [comment]
Thanks! Looks good; merging.

**Comment by chrisbobbe** [comment]
Thanks for the review!

**Comment by gnprice** [comment]
> it seems `tools/test lint` doesn't cover the files in `types/` anyway. (Not sure why, or maybe I'm missing some nuance there.)

Yeah, that's ultimately due to this bit:
```sh
# Intersect $files with the set of our JS files in src/.
#
# Prints a list of newline-terminated paths; either files, or
# directories meaning their whole subtrees.
files_js() {
    local file_pattern='^src/.*\.js$'
```

If we want to lint the type definitions, that could be accomplished by changing the `lint` case here to be like the `prettier` case:
```sh
        lint)
            run_lint $(files_js)
            ;;
        jest)
            run_jest
            ;;
        prettier)
            run_prettier $(files_js_jsflow)
            ;;
```
as `files_js_jsflow` is defined to apply to those too.

I'm not sure we do want lint on the type definitions, though, or at least the generated ones. When I open one of the longer files in `types/@react-navigation/`, it seems like there are a lot of lint errors that wouldn't be productive to try to resolve.

**Pull request merged**